### PR TITLE
fix(fslib): export `CustomDir`

### PR DESCRIPTION
**Pull request by bruce-one**
**What's the problem this PR addresses?**

I'm in the process of upgrading an application ([nexe](/nexe/nexe)) leveraging `@yarnpkg/fslib` from 2.x to 3.x.

In version 2 we were able to import `CustomDir` to create a custom implementation of `opendir` that was usable with a `FakeFS`, however this is no longer exported so doing so is no longer possible to leverage the `CustomDir` class.

...

**How did you fix it?**

This PR exports `CustomDir` in addition to `opendir` from `opendir.ts`.

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: .yarn/versions/fefcfcf9.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+releases:
```
Comment (RDIL) [inline .yarn/versions/fefcfcf9.yml:1]: Only fslib should be versioned here, the rest should be declined.

**Review by RDIL — CHANGES_REQUESTED** [review]

File: .yarn/versions/fefcfcf9.yml (hunk 1 above)
Comment (bruce-one) [reply .yarn/versions/fefcfcf9.yml:1]: Thanks :-) Sorry about that, wasn't quite sure :-)

Fixed now :-)

**Review by merceyz — APPROVED** [review]

**Review by RDIL — APPROVED** [review]

**Comment by bruce-one** [comment]
Thanks for the reviews @RDIL and @merceyz :-)

Do I need to do anything else to get this merged? No worries if not, just confirming :-)
Thanks :-)

**Comment by RDIL** [comment]
For whatever reason the deploy preview for the website is failing so we can't merge it due to the permission setup.

**Pull request merged**